src/commit_list.c

```diff
@@ -107,75 +110,34 @@ git_commit_list_node *git_commit_list_pop(git_commit_list **stack)
 
 static int commit_quick_parse(
 	git_revwalk *walk,
-	git_commit_list_node *commit,
-	const uint8_t *buffer,
-	size_t buffer_len)
+	git_commit_list_node *node,
+    git_odb_object *obj)
 {
-	const size_t parent_len = strlen("parent ") + GIT_OID_HEXSZ + 1;
-	const uint8_t *buffer_end = buffer + buffer_len;
-	const uint8_t *parents_start, *committer_start;
-	int i, parents = 0;
-	int64_t commit_time;
-
-	buffer += strlen("tree ") + GIT_OID_HEXSZ + 1;
-
-	parents_start = buffer;
-	while (buffer + parent_len < buffer_end && memcmp(buffer, "parent ", strlen("parent ")) == 0) {
-		parents++;
-		buffer += parent_len;
-	}
+	int error;
+	size_t i;
+	git_oid *parent_oid;
+	git_commit *commit;
 
-	commit->parents = alloc_parents(walk, commit, parents);
-	GITERR_CHECK_ALLOC(commit->parents);
+	if ((error = git_object__init_with_odb((git_object **)&commit, obj, walk->repo)) < 0)
+		return error;
 
-	buffer = parents_start;
-	for (i = 0; i < parents; ++i) {
-		git_oid oid;
+	if ((error = git_commit__parse_commit(commit, obj, GIT_COMMIT_PARSE_PARENTS|GIT_COMMIT_PARSE_COMMITTER)) < 0)
```

Comment:
Ok, I guess that answers my question. Nice to see some reduction in duplicated code.

Reply:
👍